scripts/cli/TimerMode.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiTest\scripts\cli;
+
+use oat\oatbox\extension\script\ScriptAction;
+use oat\oatbox\reporting\Report;
+
+class TimerMode extends ScriptAction
+{
+    public const SET_TIMER_MODE = 'set';
+    public const SUPPORTED_MODES = [
+        'server',
+        'client'
+    ];
+
+    protected function provideOptions()
+    {
+        return [
+            self::SET_TIMER_MODE => [
+                'prefix' => 's',
+                'longPrefix' => 'set',
+                'cast' => 'string',
+                'flag' => false,
+                'required' => false,
+                'description' => 'Set the timer mode',
+            ],
+        ];
+    }
+
+    protected function provideDescription()
+    {
+        return 'This command sets the test runner timer mode';
+    }
+
+    protected function run()
+    {
+        $report = Report::createInfo(sprintf('The test runner timer mode is currently set to "%s"', $this->getTimerMode()));
+
+        $mode = $this->getOption(self::SET_TIMER_MODE);
+        if ($mode) {
```

Comment:
Early returns are preferable as they decrease the indentation level in the code thus making it more readable.

```
$report = Report::createInfo(sprintf('The test runner timer mode is currently set to "%s"', $this->getTimerMode()));

$mode = $this->getOption(self::SET_TIMER_MODE);

try {

    $this->validate($mode);

    $this->changeMode($mode);
                
    $report->add(
        Report::createSuccess(sprintf('The test runner timer mode was successfully set to "%s"', $mode))
    );
} catch (Exception $exception) {
    $report->add(
        Report::createError(sprintf('Failed to change the timer mode: %s', $exception->getMessage()))
    );
}

return $report;
```

For the validate method:

```
private function validate($mode): void
{
    if (empty($mode)) {
        throw new InvalidArgumentException('The timer mode must be provided');
    }

    if (!in_array($mode, self::SUPPORTED_MODES, true)) {
        throw new InvalidArgumentException('The timer mode must be one of: ' . implode(', ', self::SUPPORTED_MODES));
    }
}
```

The rest for the change method can stay the same.